packages/jest/src/create-serializer.js

```diff
@@ -49,27 +50,14 @@ function deepTransform(node, transform) {
 
   const transformed: any = transform(node)
 
-  if (transformed !== node) {
-    if (transformed.props) {
```

Comment:
I can't say that I understand the original issue in full, but I've found out that just removing this fixes the problem and no other tests fail, so... let's roll with that.